### PR TITLE
Fix loading too many images on nodes generates glitches

### PR DESCRIPTION
**Pull request by boogheta**
This solves #1257 by retaking within @kaij's PR #1206 only the code part related to storing images in canvas memory and apply it to node.image directly

Co-authored-by: Kai Jauslin <kj@nextension.com>

## Pull request type

Check the type of change your PR introduces:

- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

### CODE REVIEW

**Comment by boogheta** [comment]
More tests on phones and tablets revealed it was not enough for some configs, I changed the max value of 2048 to 3072 as it seems to handle more devices as such

File: src/rendering/webgl/programs/node.image.ts [hunk 1]
```diff
@@ -104,41 +120,95 @@ export default function getNodeImageProgram(): typeof AbstractNodeImageProgram {
...
-        dx = (image.width - image.height) / 2;
-      } else {
-        dy = (image.height - image.width) / 2;
+      pendingImages.forEach(({ id, image, size }) => {
+        const imageSizeInTexture = Math.min(MAX_TEXTURE_SIZE, size);
+
+        // Crop image, to only keep the biggest square, centered:
+        let dx = 0,
+          dy = 0;
+        if ((image.width || 0) > (image.height || 0)) {
```
Comment (Yomguithereal) [inline src/rendering/webgl/programs/node.image.ts:153]: This `|| 0` seems to imply `width` and `height` could not be set, in which case the `else` branch could produce a `NaN`, no?

Comment (Yomguithereal) [reply src/rendering/webgl/programs/node.image.ts:153]: Fair enough this code is not yours but the diff was misleading.

**Comment by Yomguithereal** [comment]
@boogheta what happens if the total width of images is over `MAX_WIDTH` in this scenario? Also, if images are too huge, are they cropped?

**Comment by boogheta** [comment]
I think big images are centercropped indeed, that's what I understand of @kaij's code
Regarding max_width this is not completely clear to me, but I feel like then the original bug happens again : some images further in the pipe are written over first ones and we end up with images within nodes which are composite parts of other ones instead of the desired. This is why I increased it from 2048 to 3072, although increasing it too much seems to sometimes crash the browser tab

**Comment by Yomguithereal** [comment]
Note to self: should we switch to triangle primitive for this program by default?

**Pull request merged**